microprofile/WFLY-14932_mp_reactive_messaging_kafka_api.adoc

```diff
@@ -0,0 +1,119 @@
+= API for more information and configuration of Kafka messages
+:author:            Kabir Khan
+:email:             kkhan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+With the current offering of Reactive Messaging introduced in WildFLy 23, and in WildFly 24, There is currently no way to get Kafka related information when receiving a message, nor is it possible to set to Kafka when sending a message. The most important thing we are missing is the key of the Kafka record being sent/received.
+
+SmallRye Reactive Messaging has had this functionality for a long time in its Kafka connector module, but as this mixes implementation details with public API, with no clear separation of what is what, it was decided not to introduce it until a suitable API was extracted.
+
+SmallRye Reactive Messaging has been enhanced to extract a Kafka user API in https://github.com/smallrye/smallrye-reactive-messaging/pull/1272. Note that this is a starting point, and more constructs are likely to end up in the Kafka API. As this is an external project, we have influence over what we want to add (as in I will add), but we don't have influence to stop them from adding what they see fit to add.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14932[WFLY-14932]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1666[EAP7-1666]
+* https://issues.redhat.com/browse/WFLY-14798[WFLY-14798 - Upgrade to MicroProfile Reactive Messaging 2.0]
+* https://issues.redhat.com/browse/WFLY-14800[WFLY-14800 - Modernise MicroProfile Reactive Messaging Quickstart for RM 2.0]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:{jstourac@redhat.com}[Jan Stourac]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly - will need adjusting to add the new Kafka API maven module to the module containing the Kafka Connector
+* WildFly BOMs - the new API artifact will be added to these
+* SmallRye Reactive Messaging - contains the new API Maven module
+* WildFly Quickstarts - The Reactive Messaging Quickstart should be enhanced to show the Kafka API
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
```

Comment:
Nit: AFAIK this is relevant to domain mode.

(EAP XP support policies are not relevant to this analysis.)

Reply:
@bstansberry It may be relevant for WildFly since it contains the MicroProfile stuff by default. But in case of EAP, the MP stuff is present only in XP release, which doesn't support managed domain. As such, this scenario is not expected to be tested properly in the product.

So it depends on what we want to track here in these `wildfly-proposals` and how to map it against EAP, I guess.